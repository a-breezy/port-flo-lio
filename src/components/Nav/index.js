import React from "react";

// const categories = [
// 	{
// 		name: "web dev",
// 		description:
// 			"Portfolio of web development showcasing Javascript, HTML, CSS, and Database Management skills",
// 	},
// 	{
// 		name: "photography",
// 		description:
// 			"Portfolio of photograghy showcasing film photography composition and development skills",
// 	},
// 	{
// 		name: "blog",
// 		description:
// 			"Blog writings related to developement, what I'm learning and growing with",
// 	},
// ];

function Nav({ pages, setPage }) {
	return (
		<nav>
			<ul className="flex-row">
				{/* <li className="mx-2">
						<a data-testid="about" href="#about">
							About Me
						</a>
					</li>
					<li>
						<span>Contact</span>
					</li> */}
				{pages.map((page, i) => (
					<li className="mx-1" key={page.name}>
						<span onClick={() => setPage(page.comp)}>{page.name}</span>
					</li>
				))}
			</ul>
		</nav>
	);
}

export default Nav;
