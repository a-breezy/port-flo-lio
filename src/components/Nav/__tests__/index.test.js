/* eslint-disable testing-library/prefer-screen-queries */
import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Nav from "..";

// cleanup memory after each test
afterEach(cleanup);

describe("Nav Component", () => {
	// make sure Nav renders
	it("renders", () => {
		render(<Nav />);
	});

	// snapshot of rendered Nav
	it("matches snapshot", () => {
		const { asFragment } = render(<Nav />);
		// compare fragment with expected
		expect(asFragment()).toMatchSnapshot();
	});
});

// test to see if camera emoji ðŸ“¸ is visible in <h2>
describe("emoji is visible", () => {
	it("inserts emoji into h2", () => {
		// arrange
		const { getByLabelText } = render(<Nav />);
		// assert --> evaluate whether the emoji comes up
		// eslint-disable-next-line testing-library/prefer-screen-queries
		expect(getByLabelText("camera")).toHaveTextContent("ðŸ“¸");
	});
});

// test to make sure Nav links are visible
describe("links are visible", () => {
	it("inserts text into links", () => {
		// arrange
		const { getByTestId } = render(<Nav />);
		// assert
		expect(getByTestId("link")).toHaveTextContent("Here I am");
		expect(getByTestId("about")).toHaveTextContent("About Me");
	});
});
