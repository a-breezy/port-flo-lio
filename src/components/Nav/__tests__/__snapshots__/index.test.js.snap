// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <h2>
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
          ðŸ“¸ It'sa Me!
        </span>
        Here I am
      </a>
    </h2>
    <nav>
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About Me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            web dev
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            photography
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            blog
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
